File: redding/countries/functions/mobwelfaregains.m
%%% Solve model

function [welfgain] = mobwelfaregains(param,Ctradesh,tradesh,CL,L,nobs)

global alpha sigma theta epsilon LL;

xtic = tic();

% parameters;
alpha=param(1);
theta=param(2);
epsilon=param(3);

% domestic trade share;
dtradesh=diag(tradesh);
Cdtradesh=diag(Ctradesh);

% welfare gains;
welfgain=((dtradesh./Cdtradesh).^(alpha./theta)).*((L./CL).^(1-alpha));

File: redding_py/functions/mobwelfaregains.py
import numpy as np

def mobwelfaregains(param, Ctradesh, tradesh, CL, L, nobs):
    # parameters
    alpha = param[0]
    theta = param[1]
    epsilon = param[2]

    # domestic trade share
    dtradesh = np.diag(tradesh)
    Cdtradesh = np.diag(Ctradesh)

    # welfare gains
    welfgain = (dtradesh / Cdtradesh) ** (alpha / theta) * (L / CL) ** (1 - alpha)

    return welfgain